### ISSUE TITLE
Checkout page remains blocked when closing the Google Pay modal

### ISSUE THREAD

**Issue by a-danae** [woocommerce/woocommerce-square#133]
### Describe the bug

The checkout fields and buttons remain disabled when choosing to pay with Google Pay and closing the modal without completing the payment. The shopper has to reload the page to continue placing the order.

### To reproduce

1. Enable paying with Google Pay through Square
1. As a shopper, add a product to the cart and go to the checkout page
1. Click on the Google Pay button
2. Close the modal without completing a payment
3. Notice that you can't continue placing the order. The fields and the place for order button remains disabled.

**Screenshots**

https://github.com/woocommerce/woocommerce-square/assets/41606954/848bb985-9276-4c33-a6e7-79f8dd512f7f

### Expected behavior

The checkout form must be enabled when the modal is closed. The shopper must be able to continue with their order without having to reload the page.

**Comment by peterwilsoncc** [comment]
In PR #138 I've got a fix in place that will unblock the form once the purchaser presses the cancel button on the Google Pay form.

Unfortunately it then triggers validation errors which looks ugly & are not helpful as the purchaser hasn't attempted to pay via the checkout form. I've not been able to figure out how to prevent the validation errors from showing. 

This is the state of the checkout after pressing cancel. I haven't got a gif, I'm afraid, as the form insists on showing my credit card's last four and I'd rather not place that information on the issue tracker here.

![Screen Shot 2024-05-29 at 3 52 50 pm](https://github.com/woocommerce/woocommerce-square/assets/519727/dc0e4341-061a-4aa1-a5f3-b44ac40b41d7)

**Comment by jeffpaul** [comment]
@peterwilsoncc I think if we log those form errors as a separate issue (@vikrampm1 can help there), then I think that's an acceptable experience for now so that we can get #138 moving along to resolve that worse issue

**Comment by peterwilsoncc** [comment]
Thanks for that Jeff, I've moved the pull request to 10up code review.

While testing I noticed that cancelling Google Pay on the cart (blocks) page would prevent the purchaser from clicking the Google Pay button a second time in the event they cancelled accidentally. The pull request attached to this ticket also fixes the cart page.

I've created #143 as a follow up for fixing the premature validation errors.
